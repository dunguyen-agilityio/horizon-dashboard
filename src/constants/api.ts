export const API_ENTITY = {
  COMPLEX_REPORT: 'complex-report',
  CHECK_REPORT: 'check-report',
  FINANCIAL_REPORT: 'financial-report',
  USERS: 'users',
  SIGN_IN: 'auth/local',
  SIGN_UP: 'auth/local/register',
  FORGET_PASSWORD: 'auth/forgot-password',
  RESET_PASSWORD: 'auth/reset-password',
  CONTRIBUTOR: 'contributors',
};
