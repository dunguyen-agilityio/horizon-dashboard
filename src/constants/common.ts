export const QUERY_KEY = 'q';
export const STEP_KEY = 'step';
