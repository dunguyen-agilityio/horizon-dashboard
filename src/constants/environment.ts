export const API_ENDPOINT = process.env.API_ENDPOINT!;

// Firebase
export const FIREBASE_API_KEY = process.env.NEXT_PUBLIC_FIREBASE_API_KEY;
export const FIREBASE_AUTH_DOMAIN =
  process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN;
export const FIREBASE_PROJECT_ID = process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID;
export const FIREBASE_STORAGE_BUCKET =
  process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET;
export const FIREBASE_SENDER_ID = process.env.NEXT_PUBLIC_FIREBASE_SENDER_ID;
export const FIREBASE_APP_ID = process.env.NEXT_PUBLIC_FIREBASE_APP_ID;
export const FIREBASE_MEASUREMENT =
  process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT;
export const FIREBASE_CLOUD_KEY = process.env.FIREBASE_CLOUD_KEY;
