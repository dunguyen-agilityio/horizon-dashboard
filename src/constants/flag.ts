export const FLAG = {
  'United States': '/usa.webp',
  VietNam: '/vn.webp',
};

export const FLAG_KEYS = Object.keys(FLAG) as (keyof typeof FLAG)[];
