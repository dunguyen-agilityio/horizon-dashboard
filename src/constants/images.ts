export const STARBUCKS_IMAGE = {
  COVER: '/starbucks-cover.webp',
  LOGO: '/starbucks-logo.webp',
};

export const NFT_IMAGES = {
  COVER: '/nft-cover.webp',
};

export const STARBUCKS_IMAGE = {
  COVER: '/starbucks-cover.webp',
  LOGO: '/starbucks-logo.webp',
};
