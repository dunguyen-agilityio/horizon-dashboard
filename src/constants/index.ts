export * from './common';
export * from './images';
export * from './messages';
export * from './routes';
export * from './text';
export * from './endpoint';
export * from './session';
