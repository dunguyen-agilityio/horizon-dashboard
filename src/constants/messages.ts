export const TASK_MESSAGE = {
  MIN_LENGTH: 'Title is too short. Please enter at least 6 characters.',
};
