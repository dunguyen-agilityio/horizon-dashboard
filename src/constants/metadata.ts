import type { Metadata } from 'next';

export const SIGN_IN: Metadata = {
  title: 'SignIn',
  description: 'SignIn to the app',
};

export const SIGN_UP: Metadata = {
  title: 'SignUp',
  description: 'SignUp to the app',
};

export const RESET_PASSWORD: Metadata = {
  title: 'Reset Password',
  description: 'Reset Password',
};

export const FORGET_PASSWORD: Metadata = {
  title: 'Forget Password',
  description: 'Forget Password',
};
