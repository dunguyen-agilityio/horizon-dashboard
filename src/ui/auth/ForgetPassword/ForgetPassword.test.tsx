import { act, fireEvent, render, waitFor } from '@testing-library/react';

import ForgetPasswordForm from '.';

import { AUTH_ROUTES, MESSAGES, STEP_KEY } from '@/constants';
import { useSearchParams } from 'next/navigation';

const mockPush = jest.fn();

jest.mock('next/navigation', () => ({
  useRouter: jest.fn().mockImplementation(() => ({
    push: mockPush,
  })),
  useSearchParams: jest.fn(),
  usePathname: jest.fn().mockImplementation(() => AUTH_ROUTES.FORGET_PASSWORD),
}));

describe('ForgetPassword tests', () => {
  it('Should match snapshot', () => {
    const { container } = render(<ForgetPasswordForm />);
    expect(container).toMatchSnapshot();
  });

  it('Should render correctly', async () => {
    const { getByTestId, getByLabelText, getByText } = render(
      <ForgetPasswordForm />,
    );
    (useSearchParams as jest.Mock).mockImplementation(
      () => new URLSearchParams({ step: '1' }),
    );
    expect(getByTestId('forget-password-step1')).toBeInTheDocument();

    const emailField = getByLabelText('Email');

    act(() => {
      fireEvent.change(emailField, { target: { value: 'invalid' } });
      fireEvent.click(getByTestId('continue-btn'));
    });

    await waitFor(() => {
      expect(getByText(MESSAGES.EMAIL.REGEX_MISMATCH)).toBeInTheDocument();
    });
  });

  it('mockPush triggered correctly when User typing email and click continue', async () => {
    (useSearchParams as jest.Mock).mockImplementation(
      () => new URLSearchParams({ step: '1' }),
    );
    const { getByTestId, getByLabelText } = render(<ForgetPasswordForm />);
    const emailField = getByLabelText('Email');

    act(() => {
      fireEvent.change(emailField, { target: { value: 'abc@gmail.com' } });
      fireEvent.click(getByTestId('continue-btn'));
    });

    await waitFor(() => {
      expect(mockPush).toHaveBeenNthCalledWith(
        1,
        `${AUTH_ROUTES.FORGET_PASSWORD}?${STEP_KEY}=2`,
      );
    });
  });

  it('Should render correctly', async () => {
    (useSearchParams as jest.Mock).mockImplementation(
      () => new URLSearchParams({ step: '2' }),
    );
    const { getByTestId } = render(<ForgetPasswordForm />);

    await waitFor(() => {
      expect(getByTestId('forget-password-step2')).toBeInTheDocument();
    });
  });
});
