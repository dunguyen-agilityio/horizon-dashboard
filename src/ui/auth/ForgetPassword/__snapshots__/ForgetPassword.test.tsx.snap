// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgetPassword tests Should match snapshot 1`] = `
<div>
  <div
    class="w-full sm:w-96"
  >
    <form
      class="flex flex-col gap-4"
      data-testid="forget-password-step1"
    >
      <div
        class="flex flex-col gap-2 pb-8"
      >
        <h1
          class="text-primary dark:text-white text-2xl font-bold leading-2xl leading-[56px]"
        >
          Reset your password
        </h1>
        <h2
          class="text-foreground dark:text-secondary text-medium leading-large leading-6"
        >
          Enter your email address and we will send you instructions to reset your password.
        </h2>
      </div>
      <div
        class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_12px)]"
        data-filled="true"
        data-filled-within="true"
        data-has-elements="true"
        data-has-label="true"
        data-required="true"
        data-slot="base"
      >
        <div
          class="h-full flex flex-col"
          data-slot="main-wrapper"
        >
          <div
            class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 h-12 min-h-12 rounded-large transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background"
            data-slot="input-wrapper"
            style="cursor: text;"
          >
            <label
              class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block text-foreground-500 after:content-['*'] after:text-danger after:ml-0.5 rtl:after:ml-[unset] rtl:after:mr-0.5 will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] group-data-[filled-within=true]:text-foreground group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 top-1/2 -translate-y-1/2 group-data-[filled-within=true]:left-0 left-3 rtl:left-auto rtl:right-3 text-medium group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_24px)] pe-2 max-w-full text-ellipsis overflow-hidden"
              data-slot="label"
              for="react-aria-:r0:"
              id="react-aria-:r1:"
            >
              Email
            </label>
            <div
              class="inline-flex w-full items-center h-full box-border"
              data-slot="inner-wrapper"
            >
              <input
                aria-label="Email"
                aria-labelledby="react-aria-:r0: react-aria-:r1:"
                aria-required="true"
                class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-medium group-data-[has-value=true]:text-default-foreground"
                data-slot="input"
                id="react-aria-:r0:"
                name="email"
                placeholder="Enter your email"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover text-black dark:text-white opacity-50 cursor-not-allowed bg-blue-450 dark:bg-purple-750 w-full py-7 mt-6"
        data-disabled="true"
        data-testid="continue-btn"
        disabled=""
        type="submit"
      >
        <p
          class="text-primary dark:text-white text-medium leading-large text-white font-bold leading-4"
        >
          Continue
        </p>
      </button>
    </form>
  </div>
</div>
`;
